ssc_interface_wrapper_ros2/src/converter_worker.cpp

```diff
@@ -188,88 +205,104 @@ namespace ssc_interface_wrapper{
+            {
+                vehicle_status.current_gear.gear = automotive_platform_msgs::msg::Gear::NEUTRAL;
+            }
+            else if (msg_gear.current_gear.gear == automotive_platform_msgs::msg::Gear::DRIVE)
+            {
+                vehicle_status.current_gear.gear = automotive_platform_msgs::msg::Gear::DRIVE;
+            }
+
+            // lamp/light cannot be obtain from SSC
+            // vehicle_status.lamp
+            // vehicle_status.light
 
-        // lamp/light cannot be obtain from SSC
-        // vehicle_status.lamp
-        // vehicle_status.light
+            current_status_msg_ = vehicle_status;
+            have_vehicle_status_ = true; // Set vehicle status message flag to true
 
-        current_status_msg_ = vehicle_status;
-        have_vehicle_status_ = true; // Set vehicle status message flag to true
+            // Reset flags
+            received_new_msg_ = false;
+            single_update_processed_ = !single_update_processed_;
 
+        }
+        
     }                                    
 
     void Converter::callback_for_twist_update(const automotive_platform_msgs::msg::VelocityAccelCov& msg_velocity,
                                   const automotive_platform_msgs::msg::CurvatureFeedback& msg_curvature,
                                   const automotive_platform_msgs::msg::SteeringFeedback& msg_steering_wheel)
     {
-        // current steering curvature
-        double curvature = !config_.use_adaptive_gear_ratio_ ?
-                         (msg_curvature.curvature) :
-                         std::tan(msg_steering_wheel.steering_wheel_angle/ adaptive_gear_ratio_) / config_.wheel_base_;
-
-        geometry_msgs::msg::TwistStamped twist;
-        twist.header.frame_id = BASE_FRAME_ID;
-        twist.header.stamp = msg_velocity.header.stamp;
-        twist.twist.linear.x = msg_velocity.velocity;               // [m/s]
-        twist.twist.angular.z = curvature * msg_velocity.velocity;  // [rad/s]
-        current_twist_msg_ = twist;
-
-        have_twist_ = true;
+        if(velocity_msg_exists_ && curvature_msg_exists_ && steering_msg_exists_ && (received_new_msg_  || single_update_processed_))
+        {
+            // current steering curvature
+            double curvature = !config_.use_adaptive_gear_ratio_ ?
+                            (msg_curvature.curvature) :
+                            std::tan(msg_steering_wheel.steering_wheel_angle/ adaptive_gear_ratio_) / config_.wheel_base_;
+
+            geometry_msgs::msg::TwistStamped twist;
+            twist.header.frame_id = BASE_FRAME_ID;
+            twist.header.stamp = msg_velocity.header.stamp;
+            twist.twist.linear.x = msg_velocity.velocity;               // [m/s]
+            twist.twist.angular.z = curvature * msg_velocity.velocity;  // [rad/s]
+            current_twist_msg_ = twist;
+
+            have_twist_ = true;
```

Comment:
What is this used for?

Reply:
This is from the autoware.ai code. Its being used to if the geometry_msgs::msg::TwistStamped msg should be published.
It's reset each time its published. https://github.com/usdot-fhwa-stol/carma-ssc-interface-wrapper/blob/14a05157e10dd46923068feb631a17696855a3e0/ssc_interface_wrapper_ros2/src/converter_worker.cpp#L127